src/SDK/SmartDetectorsShared/Clients/ExtendedMetricClient.cs

```diff
@@ -0,0 +1,147 @@
+            : this(tracer, new MonitorManagementClient(credentialsFactory.Create("https://management.azure.com/")))
+        {
+        }
+
+        /// <summary>
+        /// Get the resource metric definitions
+        /// </summary>
+        /// <param name="resourceUri">The Uri to the resource metrics API.
+        ///                           E.g. for queues: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Storage/storageAccounts/{storageName}/queueServices/default"</param>
+        /// <param name="cancellationToken">Cancellation Token for the async operation</param>
+        /// <returns>A <see cref="Task{TResult}"/> object that represents the asynchronous operation, returning the list of metric definitions</returns>
+        [SuppressMessage("Microsoft.Design", "CA1054:UriParametersShouldNotBeStrings", Justification = "Keeping alignment with the Microsoft.Azure.Management.Monitor.Fluent.MetricDefinitionsOperationsExtensions API")]
+        public async Task<IEnumerable<MetricDefinition>> GetResourceMetricDefinitionsAsync(string resourceUri, CancellationToken cancellationToken)
+        {
+            this.tracer.TraceInformation($"Running GetResourceMetricDefinitions with an instance of {this.GetType().Name}");
+            List<MetricDefinition> definitions = (await this.retryPolicy.RunAndTrackDependencyAsync(
+                this.tracer,
+                DependencyName,
+                "GetResourceMetricDefinitions",
+                () => this.metricClient.GetResourceMetricDefinitionsAsync(
+                    resourceUri: resourceUri,
+                    cancellationToken: cancellationToken))).ToList();
+
+            this.tracer.TraceInformation($"Running GetResourceMetricDefinitions completed. Total Definitions: {definitions.Count}.");
+            return definitions;
+        }
+
+        /// <summary>
+        /// Get the resource metric definitions, based on the resource and service (for example: if the resource is a storage account, possible services are BLOB, Queue, Table and File)
+        /// </summary>
+        /// <param name="resource">The Azure resource for which we want to fetch metric definitions</param>
+        /// <param name="azureResourceService">The Azure resource's service type</param>
+        /// <param name="cancellationToken">Cancellation Token for the async operation</param>
+        /// <returns>A <see cref="Task{TResult}"/> object that represents the asynchronous operation, returning the list of metric definitions</returns>
+        public async Task<IEnumerable<MetricDefinition>> GetResourceMetricDefinitionsAsync(ResourceIdentifier resource, ServiceType azureResourceService, CancellationToken cancellationToken)
+        {
+            string resourceFullUri = MetricClient.GetResourceFullUri(resource, azureResourceService);
+            return await this.GetResourceMetricDefinitionsAsync(resourceFullUri, cancellationToken);
```

Comment:
Call metricClient.GetResourceMetricDefinitionsAsync(resource, ..)
No need to call GetResourceFullUri here